Week 04/Lecture 08/Assignment 3/src/main/java/com/lecture8/assignment3/config/AppConfig.java

```diff
@@ -0,0 +1,36 @@
+package com.lecture8.assignment3.config;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AppConfig {
+
+    
+    @Bean
+    @Qualifier("dataSource1")
+    public DataSource dataSource1() {
+        DataSourceBuilder dataSourceBuilder = DataSourceBuilder.create();
+        dataSourceBuilder.driverClassName("com.mysql.jdbc.Driver");
```

Comment:
Database info like url, username, pass should be placed in prop file and get by @Value, not hardcode in this.

Reply:
Database configuration details have been moved to the application.properties file. The values are then injected into the DataSourceConfig class using the @Value annotation.

**Changes Made:**

- Added database configuration properties to application.properties.
- Used @Value to inject these properties into DataSourceConfig.